src/update_handler.c

```diff
@@ -33,6 +33,10 @@ static GUnixOutputStream* open_slot_device(RaucSlot *slot, int *fd, GError **err
 	GError *ierror = NULL;
 	int fd_out;
 
+	g_return_val_if_fail(slot, NULL);
+	g_return_val_if_fail(fd == NULL || *fd == NULL, NULL);
```

Comment:
The fd pointer is optional (and you depend on that in another commit), so this check is wrong. I'll remove this in a fixup.